### PR TITLE
fix(save-form-flow): fixed the bugs

### PR DESCRIPTION
**Pull request by MayaSasson**

### CODE REVIEW

**Review by yaniryos — DISMISSED** [review]

File: client/src/components/App/Content/InvestigationForm/useInvestigationForm.ts [hunk 1]
```diff
@@ -18,18 +18,16 @@ import { setCountries } from 'redux/Country/countryActionCreators';
 import useStyles from './InvestigationFormStyles';
 import { defaultTab, tabs } from './TabManagement/TabManagement';
 import { useInvestigationFormOutcome, useInvestigationFormParameters } from './InvestigationFormInterfaces';
+import { initialAddress } from 'commons/Contexts/ClinicalDetailsContext';
```
Comment (EmilyLip) [inline client/src/components/App/Content/InvestigationForm/useInvestigationForm.ts:21]: import order

File: client/src/components/App/Content/LandingPage/InvestigationTable/useInvestigationTable.ts [hunk 2]
```diff
@@ -12,6 +12,7 @@ import { setEpidemiologyNum } from 'redux/Investigation/investigationActionCreat
 
 import useStyle from './InvestigationTableStyles';
 import { useInvestigationTableOutcome } from './InvestigationTableInterfaces';
+import { setIsLoading } from 'redux/IsLoading/isLoadingActionCreators';
```
Comment (EmilyLip) [inline client/src/components/App/Content/LandingPage/InvestigationTable/useInvestigationTable.ts:15]: import order

File: client/src/components/App/Content/InvestigationForm/useInvestigationForm.ts [hunk 3]
```diff
@@ -132,12 +131,13 @@ const useInvestigationForm = (parameters: useInvestigationFormParameters): useIn
     const saveClinicalDetails = () => {
         const clinicalDetails = ({
             ...clinicalDetailsVariables.clinicalDetailsData,
-            'investigatedPatientId': investigatedPatientId,
-            'epidemioligyNumber' : epidemiologyNumber,
-            'creator' : creator,
-            'lastUpdator' : lastUpdator,
+            isolationAddress: clinicalDetailsVariables.clinicalDetailsData.isolationAddress === initialAddress ?
```
Comment (EmilyLip) [inline client/src/components/App/Content/InvestigationForm/useInvestigationForm.ts:134]: make sure the isolationAddress is fully deleted when undefined is passed

File: server/src/DBService/PersonalDetails/Query.ts [hunk 4]
```diff
@@ -68,7 +68,7 @@ query MyQuery($parentOccupation: String!) {
 `;
 
 export const GET_EDUCATION_SUB_OCCUPATION_BY_CITY = gql`
-query MyQuery($city: String!) {
+query getEducationSubOccupationsByCity($city: String!) {
     allSubOccupations(condition: {parentOccupation: "מערכת החינוך", city: $city}) {
```
Comment (EmilyLip) [inline server/src/DBService/PersonalDetails/Query.ts:72]: use const

File: server/src/ClientToDBAPI/ClinicalDetailsRoute/mainRoute.ts [hunk 5]
```diff
@@ -53,49 +50,60 @@ clinicalDetailsRoute.post('/saveClinicalDetails', (request: Request, response: R
...
-            isIsolationProblem: requestInvestigation.isIsolationProblem,
-            isIsolationProblemMoreInfo: requestInvestigation.isIsolationProblemMoreInfo,
-            isolationEndTime: requestInvestigation.isolationEndTime,
-            isolationStartTime: requestInvestigation.isolationStartTime,
-            symptomsStartTime: requestInvestigation.symptomsStartTime,
-            wasHospitalized: requestInvestigation.wasHospitalized,
-            doesHaveSymptoms: requestInvestigation.doesHaveSymptoms,
-            isolationAddress: result.data.insertAndGetAddressId.integer
+    graphqlRequest(UPDATE_INVESTIGATION, request.headers,{
+        epidemiologyNumber: requestInvestigation.epidemiologyNumber,
```
Comment (daniel7764) [inline server/src/ClientToDBAPI/ClinicalDetailsRoute/mainRoute.ts:54]: Im not fimiliar with the investigation interface/objects, but wouldn't it be more comfortable/readable composing an object dedicated for sending an investigation to update? correct me if im missing something

File: server/src/ClientToDBAPI/PersonalDetailsRoute/mainRoute.ts [hunk 6]
```diff
@@ -32,6 +32,7 @@ personalDetailsRoute.get('/educationSubOccupations', (request: Request, response
 });
 
 personalDetailsRoute.post('/updatePersonalDetails', (request: Request, response: Response) => {
+    console.log(request.body);
```
Comment (OrelKaravani) [inline server/src/ClientToDBAPI/PersonalDetailsRoute/mainRoute.ts:35]: Is it necessary?

**Review by OrelKaravani — CHANGES_REQUESTED** [review]

File: server/src/ClientToDBAPI/ClinicalDetailsRoute/mainRoute.ts (hunk 5 above)
Comment (MayaSasson) [reply server/src/ClientToDBAPI/ClinicalDetailsRoute/mainRoute.ts:54]: After checking it I used only one var as you can see

**Review by daniel7764 — APPROVED** [review]

**Review by OrelKaravani — APPROVED** [review]

**Review by EmilyLip — APPROVED** [review]

**Pull request merged**